common/datavalues/src/types/type_primitive.rs

```diff
@@ -32,20 +32,20 @@ pub struct PrimitiveDataType<
 }
 
 // typetag did not support generic impls, so we have to do this
-pub fn create_primitive_datatype<T: PrimitiveType>() -> Arc<dyn DataType> {
+pub fn create_primitive_datatype<T: PrimitiveType>() -> DataTypeImpl {
     match (T::SIGN, T::FLOATING, T::SIZE) {
-        (false, false, 1) => Arc::new(UInt8Type { _t: PhantomData }),
-        (false, false, 2) => Arc::new(UInt16Type { _t: PhantomData }),
-        (false, false, 4) => Arc::new(UInt32Type { _t: PhantomData }),
-        (false, false, 8) => Arc::new(UInt64Type { _t: PhantomData }),
+        (false, false, 1) => DataTypeImpl::UInt8(UInt8Type { _t: PhantomData }),
```

Comment:
Can we use `UInt8Type::default()` here?

Reply:
Seems it could, I did not notice that `PhantomData ` default impls.

Reply:
We can continue to improve these in other PRs if feasible.